### ISSUE TITLE
Looking for help writing custom rules

### ISSUE THREAD

**Issue by jeffshaver** [palantir/tslint#4093]
This isn't a bug report, but I didn't really know where to go to get help with this. I am trying to write custom rules for TSLint. I got a repo up and running. Wrote tests and got them passing. but when I install the repo from github using yarn (didn't want to publish until it was finished), the rule runs, but the cb for `ts.forEachChild(sourceFile, cb)` never gets run. Can anyone provide help. If so, maybe this could lead to a pr improving the custom rule documentation?

Repo with the custom rule: https://github.com/jeffshaver/tslint-import-groups
Repo that doesn't error with rule installed: https://github.com/jeffshaver/tslint-import-groups-test

Any help would be appreciated. Or if there is a better place to ask a question like this.

**Comment by johnwiseheart** [comment]
StackOverflow might have been the best place to ask this question, but there may not be a whole lot of people answering `tslint` questions on there so this might be best anyway. I'll try to take a look if I get some spare time in the next couple of days.

Have you tried running the rule without using a separate directory? Does the rule work against tests?

**Comment by jeffshaver** [comment]
Thanks alot!

And the rule has tests and they work as expected. It's only when I create a
new repo, install tslint and my rule and then in the new projects
tslint.json, extend the other repo. I have already 5 tried rulesDirectory
but that didn't work either. Please let me know if there is anything I can
do to help you help me or if there is something I can do to lessen the
burden in you. Appreciate you taking the time to help

**Comment by jeffshaver** [comment]
@johnwiseheart I figured it out. And this may just be me not understand enough. I had typescript and tslint as dependencies of my rule repo. Which is what was throwing it off. I assume that is because when i install the module in the repo that is going to use the rule, it is using a different typescript module than the one in the other repo. 

As soon as I moved tslint and typescript into devDependencies and in peerDependencies, it worked.

Sorry if I caused undue work on you or anyone else.

I am not sure if this should be documented somewhere or not. If so, I am not against putting in a PR to add it in the docs somewhere.
